src/aap_eda/wsapi/consumers.py

```diff
@@ -417,3 +435,37 @@ def get_eda_system_vault_passwords(
             )
 
         return vault_passwords
+
+    def _get_url(self, message: ActionMessage, inputs: dict) -> str:
+        if message.action not in ("run_job_template", "run_workflow_template"):
+            return ""
+        url = message.url
+        if not message.controller_job_id:
+            return url
+
+        api_url = inputs["host"]
```

Comment:
It seems that this method can be called with empty `inputs`. Should not be handled the potential absence of the "host" key?

Reply:
For the input of credential type of AAP, the `hosts` is required

Reply:
the function can still be called with empty "inputs": https://github.com/ansible/eda-server/pull/948/files#diff-179935aa48d00084c8f2722f9b007cad6d09b24425ffe4121cc75f6c37b03c42R279